packages/driver/src/cy/commands/querying/querying.ts

```diff
@@ -21,6 +160,21 @@ interface InternalContainsOptions extends Partial<Cypress.Loggable & Cypress.Tim
 }
 
 export default (Commands, Cypress, cy, state) => {
+  /*
+   * cy.get() is currently in a strange state: There are two implementations of it in this file, registered one after
+   * another. It first is registered as a command (Commands.addAll()) - but below it, we *also* add .get()
+   * via Commands.overwriteQuery(), which overwrites it.
+   *
+   * This is because other commands in the driver rely on the original .get() implementation, via
+   * `cy.now('get', selector, getOptions)`.
+   *
+   * The key is that cy.now() relies on `cy.commandFns[name]` - which addAll() sets, but overwriteQuery() does not.
+   *
+   * The upshot is that any test that relies on `cy.get()` is using the query-based implementation, but various
+   * driver commands have access to the original implementation of .get() via cy.now(). This is a temporary state
+   * of affairs while we refactor other commands to also be queries - we'll eventually be able to delete this
```

Comment:
The key is that that `Commands.add()` registers the function in `cy.commandFns[name]`, while `addQuery` doesn't. So we register it as a command, creating  `cy.get()` and `cy.commandFns.get`, then register it as a query, overriding `cy.get()`.

https://github.com/cypress-io/cypress/pull/23791/files#diff-231d74866a51e1411d172ab6ca0658f7fa712e9676f216ba797c544c47dcbf94 contains the actual fix - registering queries separately (without a wrapping promise) and then updates our other commands to use `cy.now('get')` properly when .get is a query - but I didn't have that ready when I made this PR, and I think it makes sense to merge in separately regardless (since it's a breaking change, needs to go in 11.x branch and not develop).